File: lista3/zadanie2/src/Quadrangle.cpp
#include "Quadrangle.hpp"

Quadrangle::Quadrangle(ClientInput& input)
{
    sides = readSidesFromInput(input);
    angle = input.nextInput();
}

std::set<double> Quadrangle::readSidesFromInput(ClientInput &input)
{
    std::set<double> result;
    for(int i = 0; i < 4; i++)
    {
        result.insert(input.nextInput());
    }
    return result;
}

File: lista3/zadanie1/Quadrangle.java
import java.util.*;

public abstract class Quadrangle extends Figure
{
    public Quadrangle(ClientInput input)
    {
        sides = readSidesFromInput(input);
        angle = input.NextInput();
    }

    public int ChangeInIndex()
    {
        return 5;
    }

    protected static Set<Double> readSidesFromInput(ClientInput input)
    {
        Set<Double> result = new LinkedHashSet<Double>();
        for(int i = 0; i < 4; i++)
        {
            result.add(input.NextInput());
        }
        return result;
    }

    protected Set<Double> sides;
    protected double angle;
};
